packages/frontend-2/pages/server-management/active-users.vue

```diff
@@ -0,0 +1,332 @@
+<template>
+  <div>
+    <Portal to="navigation">
+      <HeaderNavLink
+        :to="'/server-management'"
+        name="Server Management"
+      ></HeaderNavLink>
+      <HeaderNavLink
+        :to="'/server-management/active-users/'"
+        name="Active Users"
+      ></HeaderNavLink>
+    </Portal>
+
+    <h1 class="h4 font-bold mb-4">Active Users</h1>
+
+    <FormTextInput
+      size="lg"
+      name="search"
+      :custom-icon="MagnifyingGlassIcon"
+      color="foundation"
+      full-width
+      search
+      :show-clear="!!searchString"
+      placeholder="Search Users"
+      class="rounded-md border border-outline-3"
+      @update:model-value="debounceSearchUpdate"
+      @change="handleSearchChange"
+    />
+
+    <Table
+      :headers="[
+        { id: 'name', title: 'Name' },
+        { id: 'email', title: 'Email' },
+        { id: 'emailState', title: 'Email State' },
+        { id: 'company', title: 'Company' },
+        { id: 'role', title: 'Role' }
+      ]"
+      :items="users"
+      :buttons="[{ icon: TrashIcon, label: 'Delete', action: openUserDeleteDialog }]"
+      :column-classes="{
+        name: 'col-span-3',
+        email: 'col-span-3',
+        emailState: 'col-span-2',
+        company: 'col-span-2',
+        role: 'col-span-2'
+      }"
+      :overflow-cells="true"
+    >
+      <template #name="{ item }">
+        <div class="flex items-center gap-2">
+          <Avatar :user="item" />
+          {{ item.name }}
+        </div>
+      </template>
+
+      <template #email="{ item }">
+        {{ item.email }}
+      </template>
+
+      <template #emailState="{ item }">
+        <div class="flex items-center gap-2 select-none">
+          <template v-if="item.verified">
+            <ShieldCheckIcon class="h-4 w-4 text-primary" />
+            <span>verified</span>
+          </template>
+          <template v-else>
+            <ShieldExclamationIcon class="h-4 w-4 text-danger" />
+            <span>not verified</span>
+          </template>
+        </div>
+      </template>
+
+      <template #company="{ item }">
+        {{ item.company }}
+      </template>
+
+      <template #role="{ item }">
+        <UserRoleSelect
+          :model-value="item.role"
+          @update:model-value="
+            (newRoleValue) => openChangeUserRoleDialog(item as UserItem, newRoleValue)
```

Comment:
Since `item` is going to be typed like `UserItem | ProjectItem` you can introduce a type guard like this so that you don't have to forcefully recast types and can capture scenarios where `item` is unexpectedly not `UserItem`

Example:
```ts
const asUser = (val: UserItem | ProjectItem): val is UserItem => {
  if (has(val, 'userId')) return true
  throw new Error('Unexpectedly item is not a user item! Wtf, we are in active-users.vue how did this happen?')
}

// then in your vue template
(newRoleValue) => openChangeUserRoleDialog(asUser(item), newRoleValue)
```